### PR TITLE
Entity generalization

### PR DESCRIPTION
**Pull request by VISHNUDAS-tunerlabs**
for code review not for merging

### CODE REVIEW

File: generics/constants/common.js [hunk 1]
```diff
@@ -44,5 +44,8 @@ module.exports = {
     "CHILDREN" : "children",
     "ATTACHMENT_TYPE_LINK" : "link",
     "PROJECT_ATTACHMENT" : "project",
-    "TASK_ATTACHMENT" : "task"
+    "TASK_ATTACHMENT" : "task",
+    "SUNBIRD_SERVER_TIMEOUT" : 5000,
+    "TIMEOUT_ERROR" : "Server timeout error",
```
Comment (priyanka-TL) [inline generics/constants/common.js:49]: @VISHNUDAS-tunerlabs if its a response status/message  add it in generics/https-status-codes

File: generics/services/sunbird.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    result["data"] = response.result;
+                } else {
+                      result.success = false;
+                }
+            }
+            return resolve(result);
+        }
+
+        setTimeout(function () {
+           return reject (CONSTANTS.common.TIMEOUT_ERROR)
```
Comment (priyanka-TL) [inline generics/services/sunbird.js:64]: @VISHNUDAS-tunerlabs check value of success if its not passing make it false

File: module/userProjects/helper.js [hunk 3]
```diff
@@ -2479,25 +2477,42 @@ function _projectCategories(categories) {
 function _entitiesInformation(entityIds) {
     return new Promise(async (resolve, reject) => {
         try {
+            let bodyData={
+                "id" : entityIds
+            };
+           
+        
+            let entityDetails = await sunbirdService.learnerLocationSearch(bodyData);
```
Comment (priyanka-TL) [inline module/userProjects/helper.js:2485]: @VISHNUDAS-tunerlabs handle if entityDetails is empty

File: module/userProjects/helper.js [hunk 4]
```diff
@@ -2479,25 +2477,42 @@ function _projectCategories(categories) {
...
+        
+            let entities = entityDetails.data.response;
 
-            let entityData =
-                await coreService.entityDocuments(
-                    entityIds,
-                    ["metaInformation", "entityType", "entityTypeId", "registryDetails"]
-                );
-
-            if (!entityData.success) {
```
Comment (priyanka-TL) [inline module/userProjects/helper.js:2489]: @VISHNUDAS-tunerlabs check with success value. what will be the value of "success" when API triggering timeout error

File: generics/services/sunbird.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        }, CONSTANTS.common.SUNBIRD_SERVER_TIMEOUT);
```
Comment (priyanka-TL) [inline generics/services/sunbird.js:65]: @VISHNUDAS-tunerlabs if its a api response message add it in generics/api-responses

File: envVariables.js [hunk 6]
```diff
@@ -35,6 +35,10 @@ let enviromentVariables = {
   "KAFKA_URL" : {
     "message" : "Required",
     "optional" : false
+  },
+  "SUNBIRD_SERVICE_URL" : {
```
Comment (priyanka-TL) [inline envVariables.js:39]: @VISHNUDAS-tunerlabs already USER_SERVICE_URL is there in code use that. we have added this for profile read API

File: generics/services/sunbird.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
+/**
```
Comment (priyanka-TL) [inline generics/services/sunbird.js:1]: @VISHNUDAS-tunerlabs already we have file for user/read api use that file

File: module/userProjects/helper.js [hunk 8]
```diff
@@ -21,6 +21,7 @@ const projectTemplateTaskQueries = require(DB_QUERY_BASE_PATH + "/projectTemplat
 const kafkaProducersHelper = require(GENERICS_FILES_PATH + "/kafka/producers");
 const removeFieldsFromRequest = ["submissionDetails"];
 const programsQueries = require(DB_QUERY_BASE_PATH + "/programs");
+const sunbirdService = require(GENERICS_FILES_PATH + '/services/sunbird');
```
Comment (priyanka-TL) [inline module/userProjects/helper.js:24]: @VISHNUDAS-tunerlabs use user service

File: module/userProjects/helper.js [hunk 9]
```diff
@@ -1395,7 +1394,7 @@ module.exports = class UserProjectsHelper {
                     }
                     createProject =
                         _.merge(createProject, programAndSolutionInformation.data);
-                } else {
+                } else{
```
Comment (priyanka-TL) [inline module/userProjects/helper.js:1397]: @VISHNUDAS-tunerlabs format the code

File: module/userProjects/helper.js [hunk 10]
```diff
@@ -2479,25 +2477,42 @@ function _projectCategories(categories) {
...
+            let entityDetails = await sunbirdService.learnerLocationSearch(bodyData);
 
-            let entityData =
-                await coreService.entityDocuments(
-                    entityIds,
-                    ["metaInformation", "entityType", "entityTypeId", "registryDetails"]
-                );
-
-            if (!entityData.success) {
+            if (!entityDetails.success || !entityDetails.data || !entityDetails.data.response.length > 0) {
```
Comment (priyanka-TL) [inline module/userProjects/helper.js:2487]: @VISHNUDAS-tunerlabs check entityDetails.data.response will exist always

File: package.json [hunk 11]
```diff
@@ -31,6 +31,8 @@
     "bunyan": "^1.8.12",
     "bunyan-format": "^0.2.1",
     "cache-manager": "^3.1.0",
+    "chai": "^4.2.0",
```
Comment (priyanka-TL) [inline package.json:34]: @VISHNUDAS-tunerlabs why chai & mocha added to this file? are we added any test case

File: package.json [hunk 12]
```diff
@@ -49,21 +51,20 @@
...
     "p-each-series": "^2.1.0",
     "path": "^0.12.7",
     "request": "^2.88.2",
     "require-all": "^3.0.0",
     "uuid": "^8.3.0",
-    "xml-js": "^1.6.11",
-    "chai": "^4.2.0",
-    "chai-http": "^4.3.0",
-    "mocha": "^6.2.2"
+    "xml-js": "^1.6.11"
```
Comment (priyanka-TL) [inline package.json:67]: @VISHNUDAS-tunerlabs why xml-js package? where we are using?

File: package.json (hunk 11 above)
Comment (VISHNUDAS-tunerlabs) [reply package.json:34]: also present in master

**Pull request merged**